File: Java/ToyLanguageGUI_exam_final/src/domain/expressions/ConstExpression.java
package domain.expressions;

import utils.IDictionary;
import utils.IHeap;

public class ConstExpression implements Expression {
    private int constant;
    public ConstExpression(int value){
        constant = value;
    }

    @Override
    public int eval(IDictionary<String,Integer> symbTable, IHeap<Integer,Integer> heap){
        return constant;
    }

    @Override
    public String toString(){
        return "" + constant;
    }
}

File: ToyInterpretor/ToyInterpretor/Domain/Expressions/ConstExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyInterpretor.Utils;


namespace ToyInterpretor.Domain.Expressions
{
    class ConstExpression : Expression
    {
        private int constant;

        public ConstExpression(int val)
        {
            constant = val;
        }

        public int eval(ISymTable<String, int> symb)
        {
            return constant;
        }

        public override string ToString()
        {
            return ""+constant;
        }
    }
}
